zerver/models.py

```diff
@@ -1053,6 +1054,27 @@ def parse_usermessage_flags(val):
         mask <<= 1
     return flags
 
+class Attachment(models.Model):
+    MAX_FILENAME_LENGTH = 100
+    file_name = models.CharField(max_length=MAX_FILENAME_LENGTH, db_index=True)
+    path_id = models.TextField(default='/some/path')
+    owner = models.ForeignKey(UserProfile)
+    message = models.ForeignKey(Message, blank=True, null=True)
+    is_claimed = models.BooleanField(default=False)
+
+    def __repr(self):
+        return (u"<Attachment: %s>" % (self.file_name))
+
+@cache_with_key(attachments_by_owner_id_cache_key, timeout=3066*24*7)
+def get_attachments_by_owner_id(uid):
+    return Attachment.objects.filter(owner=uid).select_related('owner')
+
+@cache_with_key(owners_from_file_name_cache_key, timeout=3066*24*7)
+def get_owners_from_file_name(file_name):
+    #The returned vaule will list of owners since different users can upload
+    #same file
+    return Attachment.objects.filter(file_name=file_name).select_related('owner')
```

Comment:
I don't think this function is a good fit for caching; people don't upload/download files that often, so I think we should just let the database handle those queries.